### DISCUSSION TITLE
Is there any way to get Content Version

### DISCUSSION THREAD

**Discussion by AmadeuAntunes** [OrchardCMS/OrchardCore#8307] [General]
https://orcharddojo.net/orchard-resources/Library/Wiki/Versioning
I noticied that in db the table [dbo].[Document] already as a version field but seems that show "1" even when I had a published version and a draft version. 
Is there any way to has a version friendly number in orchard?

**Comment by Skrypt** [comment]
What do you mean by version friendly number?

**Reply by AmadeuAntunes** [reply]
I would like to use the number of version like 1 , 2 ,3 etc but seems that the version field in database isn't exactly what I was looking.
Every time I create a draft or new public document it seems generate a new entry in database and also a ContentItemVersionId, however I'm looking for just the version number and the ContentItemVersionId isn't a friendly data to display.

**Reply by Skrypt** [reply]
You can order these ContentItems by CreatedUtc and display a number which is not related to anything technical?

int i = 1
loop
i++ ...

**Reply by jtkech** [reply]
@AmadeuAntunes @Skrypt 

For info, the `Version` number is used for an optional concurrency checking when saving a document, if used when saving a document we increment it but we check that the existing value it is still equal to the previous one.

We already use it for some document as `SiteSettings`, but not for content items whose `Version` is always equal to 1

@deanmarcussen just saw your answer ;)

**Reply by jtkech** [reply]
Yes the `ContentItemVersionId` is not incremental, it's a uid, otherwise you have the `Id` and `DocumentId` that are incremental but not per document ;)

**Comment by deanmarcussen** [comment]
The `Version` property you are seeing in the database is used internally for ConcurrencyChecking.

It's not a `ContentItemVersion` number.

The `ContentItemVersionId` is a string guid, and we don't track a count of the number of versions anywhere.

If you want to have some sequential versioning for display in the front end ui somewhere for some reason, you could use a number field, and increment the value in a content handler when a new version is published.

Or use a date, or just use the `ContentItemVersionId`

**Comment by Skrypt** [comment]
Should be a combination of ContentItemVersionId + CreatedUtc to be able to sort by date to keep the proper historical order.
There's nothing as a Version (incremental) number which is generally only usefull for display.

Question is then, what you need a Version (incremental) number for? Explain your use case in details. Else, I think the data we have is appropriate for doing almost everything.

**Reply by jtkech** [reply]
@Skrypt 

> Should be a combination of ContentItemVersionId + CreatedUtc

Good idea but here would be `ContentItemId`

Maybe on YesSql side we could just always increment the `Version`, even for a document that we save without the concurrency checking

But as @Skrypt said, you can just order by one of the dates `CreateUtc` / `ModifedUtc` / `PublishedUtc` and then having a "version" number equals to the position in the collection

**Comment by hyzx86** [comment]
@dalenewman   Thank you for your explanation ，I also always thought that Version property was the Version number..

@Skrypt 
I planned to use this field to query the audit audit log.
If used for an e-commerce project, we can use it to store a snapshot of product information at the time of purchase

**Reply by Skrypt** [reply]
In this case, this is still risky. You should at least keep a copy of the minimum required details (data) of the product in the "invoice" object persisted.
